AppSyncRealTimeClient/ConnectionProvider/ConnectionProviderError.swift

```diff
@@ -23,6 +23,9 @@ public enum ConnectionProviderError: Error {
     /// payload in dictionary format.
     case subscription(String, [String: Any]?)
 
-    /// Any other error is identified by this type
-    case other
+    /// Caused when not authorized to establish the connection.
+    case unauthorized
+
+    /// Unknown error
+    case unknown(message: String? = nil, causedBy: Error? = nil, payload: [String: Any]?)
```

Comment:
revisting this.. the payload comes from the response, so it's like bubbling up the response from the connection request. I think it's valuable to pass it back, but not sure if we need to encasuplate it in an Error object

Reply:
i think having the two associated types (error and payload) makes sense since unhandled/unclassified scenarios may occur anywhere, in which we can bubble up the information as is- whether it's the payload or an error object